wallet/src/main/kotlin/net/thechance/wallet/service/StatementService.kt

```diff
@@ -38,13 +38,15 @@ class StatementService(
             types = types,
             startDateTime = startDateTime,
             endDateTime = endDateTime,
-            openingBalance = getOpeningBalance(userId, startDate),
-            closingBalance = getClosingBalance(userId, endDate)
+            openingBalance = getOpeningBalance(userId, startDateTime),
+            closingBalance = getClosingBalance(userId, endDateTime)
         )
     }
 
     private fun getStartDateTime(startDate: LocalDate?, userId: UUID): LocalDateTime {
-        return startDate?.atStartOfDay() ?: transactionService.getUserFirstTransactionDate(userId).orNow()
+        val dateTime = startDate?.atStartOfDay()
+            ?: transactionService.getUserFirstTransactionDate(userId).orNow()
+        return dateTime.toLocalDate().atStartOfDay()
```

Comment:
why you convert it to date and take the start of the date as date time ?

Reply:
To include all the transaction in this date from the beginning of the day as this is the start date and for the end date it's at the end of the day.

Reply:
```suggestion
        return startDate?.atStartOfDay()
            ?: transactionService.getUserFirstTransactionDate(userId)
                .orNow().toLocalDate().atStartOfDay()
```
I suggest to make it like this t avoid multiple mapping if the startDate not null

Reply:
Ok makes sense